x-pack/plugins/ml/public/data_frame/common/pivot_group_by.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Dictionary } from '../../../common/types/common';
+
+export enum PIVOT_SUPPORTED_GROUP_BY_AGGS {
+  DATE_HISTOGRAM = 'date_histogram',
+  HISTOGRAM = 'histogram',
+  TERMS = 'terms',
+}
+
+type FieldName = string;
+
+interface GroupByConfigBase {
+  field: FieldName;
+  formRowLabel: string;
+}
+
+// Don't allow an interval of '0', but allow a float interval of '0.1' with a leading zero.
+export const histogramIntervalFormatRegex = /^([1-9][0-9]*((\.)([0-9]+))?|([0](\.)([0-9]+)))$/;
```

Comment:
It is still accepting input like `0.0`

Reply:
Improved the interval checking and added test coverage in 0b92f7b.